src/languages/mathematica.js

```diff
@@ -1,59 +1,66 @@
 /*
-Language: Mathematica
-Description: Wolfram Mathematica (usually termed Mathematica) is a modern technical computing system spanning most areas of technical computing.
-Authors: Daniel Kvasnicka <dkvasnicka@vendavo.com>, Jan Poeschko <jan@poeschko.com>
+Language: Wolfram Language
+Description: The Wolfram Language is the programming language used in Wolfram Mathematica, a modern technical computing system spanning most areas of technical computing.
+Authors: Patrick Scheibe <patrick@halirutan.de>, Robert Jacobson <robertjacobson@acm.org>
 Website: https://www.wolfram.com/mathematica/
 Category: scientific
 */
 
 export default function(hljs) {
+  /*
+  This dangerously looking beast of a regex was carefully assembled by Robert Jacobson.
+  See: https://wltools.github.io/LanguageSpec/Specification/Syntax/Number-representations/
+   */
+  const NUMBERS = {
+    className: 'number',
+    begin: '(?:(?:(?:[2-9]|[1-2]\\d|[3][0-5])\\^\\^(?:\\w*\\.\\w+|\\w+\\.\\w*|\\w+))|(?:\\d*\\.\\d+|\\d+\\.\\d*|\\d+))' +
```

Comment:
This would definitely be welcome but it should be broken out into `variants` and commented for readability.